doc/ipmc/IPMC_Data_Plane_HLD.md

```diff
@@ -0,0 +1,472 @@
+    - [SAI API](#sai-api)
+    - [Configuration and management](#configuration-and-management)
+      - [Manifest](#manifest)
+      - [CLI/YANG model Enhancements](#cliyang-model-enhancements)
+        - [Interface Yang Model](#interface-yang-model)
+        - [IPMC commands](#ipmc-commands)
+          - [Config commads](#config-commads)
+          - [Show commands](#show-commands)
+      - [Config DB Enhancements](#config-db-enhancements)
+    - [Warmboot and Fastboot Design Impact](#warmboot-and-fastboot-design-impact)
+    - [Restrictions/Limitations](#restrictionslimitations)
+    - [Testing Requirements/Design](#testing-requirementsdesign)
+      - [Unit Test cases](#unit-test-cases)
+      - [System Test cases](#system-test-cases)
+    - [Open/Action items](#openaction-items)
+
+### Revision
+
+|  Rev  |  Date            |  Author        | Change Description |
+| :---  | :--------------- | :------------- | :----------------  |
+|  0.1  |  Aug-13-2024     | Hb-micas       | Initial version    |
+
+### Scope
+
+This document describes how to implement the IPMC data plane feature in SONiC infrastructure.
+
+- This document only discusses the SONiC data plane integration with IPMC functionality
+- Other multicast routing protocols that use the PIM protocol (or other network protocols) are beyond the scope of this document
+
+### Definitions/Abbreviations
+
+Table 1: Abbreviations
+
+| Abbreviation    | Description                                        |
+| :----------     | :------------------------------------------------  |
+| FRR             | Free Range Routing Stack                           |
+| CLI             | Command Line Interface                             |
+| ASM             | Any-Source Multicast                               |
+| SSM             | Source-Specific Multicast                          |
+| RPF             | Reverse Path Forwarding                            |
+| IPMC            | IP Multicast                                       |
+| Mroute          | Multicast Route                                    |
+| Nhg             | NextHop Group                                      |
+
+### Overview
+
+IP multicast is a network communication technique that allows a single sender to send packets to multiple destinations without having to send packets separately for each destination. This method greatly saves bandwidth resources and improves the efficiency of the network. For different Multicast group members, Multicast service models can be divided into ASM(Any-Source multicast) and SSM(Source-Specific multicast) service models. To ensure efficient transmission of multicast data, IP multicast uses the RPF (Reverse Path Forwarding) mechanism.
+
+Routers dynamically establish forwarding tables through protocols (such as IGMP, PIM, MSDP, etc.) and maintain a multicast forwarding table, recording the membership of the multicast group and the corresponding outbound interface.
+

```

Comment:
Can you request a slot to review this HLD in Routing WG? FYI. @eddieruan-alibaba 
Few questions from the community meeting today.
1) Why didnt we explore fpmsyncd path for programming the multicast routes? we came to know pimd is directly programming the kernel, what if we have the bgp ipv4 multicast configuration, wont there be any MRTM (part of Zebra)? we need to discuss with FRR folks and decide.

Reply:
Can you describe your use case as well? 

I am second on @venkatmahalingam 's comment. It is better to explore fpmsyncd approach to program mroutes directly from pimd instead of via Linux kernel for the following two reasons.

1. Scale and performance
2. feature velocity 

Both of these two considerations are related to your use case and roadmap.

Reply:
No problem, next we will try to request Routing WG review of the HLD. As for choosing between fpmsyncd or kernel, both solutions have their own significant advantages, so it's hard for me to decide :) Let's leave this discussion for after communicating with Routing WG and see what their feedback is